### PR TITLE
Adding console tab VMI page - VNC console

### PR DESCRIPTION
**Pull request by metalice**
Signed-off-by: Matan Schatzman <mschatzm@redhat.com>

## 📝 Description

> Add a brief description

## 🎥 Demo
image

### CODE REVIEW

**Comment by metalice** [comment]
@glekner @avivtur  @yaacov @pcbailey @vojtechszocs @hstastna @upalatucci please review :)

File: src/views/virtualmachinesinstance/details/tabs/console/components/vnc-console/utils/VNCConsoleConsts.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export enum VNCState {
+  'connecting' = 'connecting',
```
Comment (avivtur) [inline src/views/virtualmachinesinstance/details/tabs/console/components/vnc-console/utils/VNCConsoleConsts.ts:2]: shouldn't this be without the single quotes?

File: src/views/virtualmachinesinstance/details/tabs/console/components/vnc-console/VncConsole.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  const [rfb, setRfb] = React.useState<any>();
+  const [status, setStatus] = React.useState<VNCState>(disconnected);
+  const staticRenderLocaitonRef = React.useRef(null);
+  const StaticRenderLocaiton = React.useMemo(
+    () => <div id={consoleContainerId} ref={staticRenderLocaitonRef} />,
+    [staticRenderLocaitonRef, consoleContainerId],
+  );
+
+  const url = React.useMemo(
+    () => `${encrypt ? 'wss' : 'ws'}://${host}:${port}/${path}`,
```
Comment (avivtur) [inline src/views/virtualmachinesinstance/details/tabs/console/components/vnc-console/VncConsole.tsx:58]: maybe put `ws` and `wss` as consts in `VirtualMachinesInstancePageConsoleTabConsts.ts`?

**Review by avivtur — COMMENTED** [review]
looks good overall

File: src/views/virtualmachinesinstance/details/tabs/console/components/vnc-console/utils/VNCConsoleConsts.ts (hunk 1 above)
Comment (upalatucci) [reply src/views/virtualmachinesinstance/details/tabs/console/components/vnc-console/utils/VNCConsoleConsts.ts:2]: without single quote is:
```js
export enum VNCState {
  connecting,
}
```

Comment (metalice) [reply src/views/virtualmachinesinstance/details/tabs/console/components/vnc-console/utils/VNCConsoleConsts.ts:2]: actually enum in typescript in this way:
export enum VNCState {
  connecting,
dissocnnected
}

will provide values of connecting = 0 , disconnected = 1 , which is a number, the VNCState is string base, so I cannot use the short version.

**Pull request merged**